app/src/main/java/com/keylesspalace/tusky/adapter/AccountAdapter.kt

```diff
@@ -0,0 +1,92 @@
+/* Copyright 2017 Andrew Dawson
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+package com.keylesspalace.tusky.adapter
+
+import androidx.recyclerview.widget.RecyclerView
+import com.keylesspalace.tusky.entity.Account
+import com.keylesspalace.tusky.interfaces.AccountActionListener
+import com.keylesspalace.tusky.util.removeDuplicates
+
+/** Generic adapter with bottom loading indicator. */
+abstract class AccountAdapter internal constructor(
+    var accountActionListener: AccountActionListener,
+    protected val animateAvatar: Boolean,
+    protected val animateEmojis: Boolean
+) : RecyclerView.Adapter<RecyclerView.ViewHolder?>() {
+    var accountList = mutableListOf<Account>()
+    private var bottomLoading: Boolean = false
+
+    override fun getItemCount(): Int {
+        return accountList.size + if (bottomLoading) 1 else 0
+    }
+
+    override fun getItemViewType(position: Int): Int {
+        return if (position == accountList.size && bottomLoading) {
+            VIEW_TYPE_FOOTER
+        } else {
+            VIEW_TYPE_ACCOUNT
+        }
+    }
+
+    fun update(newAccounts: List<Account>) {
+        accountList = removeDuplicates(newAccounts)
```

Comment:
Kotlin has a built in function for this https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/distinct.html

Reply:
It produces immutable list and we need a mutable one and if I re-implement it or add toMutableList() it won't be simpler